src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -177,12 +179,19 @@ impl<S: StateStore> StorageTable<S> {
         assert_eq!(order_types.len(), pk_indices.len());
 
         let (output_columns, output_indices) = find_columns_by_ids(&table_columns, &column_ids);
-        assert!(
-            output_indices.iter().all(|i| value_indices.contains(i)),
-            "output_indices must be a subset of value_indices"
-        );
+        let mut output_indices_in_value = vec![];
+        let mut output_indices_in_key = vec![];
+
+        for idx in output_indices {
+            if value_indices.contains(&idx) {
+                output_indices_in_value.push(idx);
+            } else {
+                output_indices_in_key.push(idx);
+            }
+        }
+
         let schema = Schema::new(output_columns.iter().map(Into::into).collect());
-        let mapping = ColumnMapping::new(output_indices);
+        let mapping = ColumnMapping::new(output_indices_in_value);
```

Comment:
Suppose `output_indices` is `[0, 1, 2]` and `value_indices` is `[2]`, we will get `output_indices_in_value` = `[2]`. Using `[2]` to project the value (its length = 1) seems incorrect.

Reply:
Indeed, I will fix it soon. Thank you.